zoomto-window.cpp

```diff
@@ -0,0 +1,290 @@
+//_/_/
+//_/_/ AERA Visualizer
+//_/_/ 
+//_/_/ Copyright (c) 2018-2022 Jeff Thompson
+//_/_/ Copyright (c) 2018-2022 Kristinn R. Thorisson
+//_/_/ Copyright (c) 2018-2022 Icelandic Institute for Intelligent Machines
+//_/_/ Copyright (c) 2021 Karl Asgeir Geirsson
+//_/_/ http://www.iiim.is
+//_/_/
+//_/_/ --- Open-Source BSD License, with CADIA Clause v 1.0 ---
+//_/_/
+//_/_/ Redistribution and use in source and binary forms, with or without
+//_/_/ modification, is permitted provided that the following conditions
+//_/_/ are met:
+//_/_/ - Redistributions of source code must retain the above copyright
+//_/_/   and collaboration notice, this list of conditions and the
+//_/_/   following disclaimer.
+//_/_/ - Redistributions in binary form must reproduce the above copyright
+//_/_/   notice, this list of conditions and the following disclaimer 
+//_/_/   in the documentation and/or other materials provided with 
+//_/_/   the distribution.
+//_/_/
+//_/_/ - Neither the name of its copyright holders nor the names of its
+//_/_/   contributors may be used to endorse or promote products
+//_/_/   derived from this software without specific prior 
+//_/_/   written permission.
+//_/_/   
+//_/_/ - CADIA Clause: The license granted in and to the software 
+//_/_/   under this agreement is a limited-use license. 
+//_/_/   The software may not be used in furtherance of:
+//_/_/    (i)   intentionally causing bodily injury or severe emotional 
+//_/_/          distress to any person;
+//_/_/    (ii)  invading the personal privacy or violating the human 
+//_/_/          rights of any person; or
+//_/_/    (iii) committing or preparing for any act of war.
+//_/_/
+//_/_/ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND 
+//_/_/ CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, 
+//_/_/ INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
+//_/_/ MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+//_/_/ DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR 
+//_/_/ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+//_/_/ SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+//_/_/ BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+//_/_/ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+//_/_/ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+//_/_/ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+//_/_/ NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+//_/_/ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY 
+//_/_/ OF SUCH DAMAGE.
+//_/_/ 
+//_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/_/
+
+
+#include "aera-visualizer-window.hpp"
+#include "graphics-items/aera-visualizer-scene.hpp"
+#include "zoomto-window.hpp"
+
+#include <regex>
+#include <string>
+#include <string.h>
+
+using namespace std;
+
+
+namespace aera_visualizer {
+
+  ZoomToWindow::ZoomToWindow(AeraVisualizerWindow *parent, ReplicodeObjects& replicodeObjects) {
```

Comment:
If you add `: QDialog(parent)` then the dialog will not have a separate entry on the Windows task bar (which is what we want, I think.) See: https://stackoverflow.com/a/40240764

Reply:
Okay, sounds reasonable. This changes the behavior a little bit since the Zoom To window now stays on top while it's open, but I think that makes sense for this kind of Find feature. You can still use the main window as normal, it'll just sit there and give an overview of the search progress I suppose.